modules/openapi-generator/src/main/resources/powershell-experimental/http_signature_auth.mustache

```diff
@@ -0,0 +1,162 @@
+{{>partial_header}}
+<#
+.SYNOPSIS
+Get the API key Id and API key file path.
+
+.DESCRIPTION
+Get the API key Id and API key file path. If no api prefix is provided then it use default api key prefix 'Signature'
+.OUTPUTS
+PSCustomObject : This contains APIKeyId, APIKeyFilePath, APIKeyPrefix
+#>
+function Get-{{{apiNamePrefix}}}APIKeyInfo {
+    $ApiKeysList = $Script:Configuration['ApiKey']
+    $ApiKeyPrefixList = $Script:Configuration['ApiKeyPrefix']
+    $apiPrefix = "Signature"
+
+    if ($null -eq $ApiKeysList -or $ApiKeysList.Count -eq 0) {
+        throw "Unable to reterieve the api key details"
+    }
+
+    if ($null -eq $ApiKeyPrefixList -or $ApiKeyPrefixList.Count -eq 0) {
+        Write-Verbose "Unable to reterieve the api key prefix details,setting it to default ""Signature"""
+    }
+
+    foreach ($item in $ApiKeysList.GetEnumerator()) {
+        if (![string]::IsNullOrEmpty($item.Name)) {
+            if (Test-Path -Path $item.Value) {
+                $apiKey = $item.Value
+                $apikeyId = $item.Name
+                break;
+            }
+            else {
+                throw "API key file path does not exist."
+            }
+        }
+    }
+
+    if ($ApiKeyPrefixList.ContainsKey($apikeyId)) {
+        $apiPrefix = ApiKeyPrefixList[$apikeyId]
+    }
+
+    if ($apikeyId -and $apiKey -and $apiPrefix) {
+        $result = New-Object -Type PSCustomObject -Property @{ 
+            ApiKeyId       = $apikeyId;
+            ApiKeyFilePath = $apiKey
+            ApiKeyPrefix   = $apiPrefix
+        }
+    }
+    else {
+        return $null
+    }
+    return $result
+}
+
+<#
+.SYNOPSIS
+    Gets the headers for http signed auth.
+
+.DESCRIPTION
+    Gets the headers for the http signed auth. It use (targetpath), date, host and body digest to create authorization header.
```

Comment:
The HTTP signature spec does not mandate this specific sequence of HTTP headers. The headers are supposed to be provided by the caller.